alf/bin/train.py

```diff
@@ -239,23 +292,48 @@ def main(_):
             # in different work processes.
             manager = mp.Manager()
             paras_queue = manager.Queue()
-            with common.get_unused_port(12355) as port:
+            with common.get_unused_port(12360) as port:
                 # The other process will communicate with the authoritative
                 # process via network protocol on localhost:port.
                 os.environ['MASTER_PORT'] = str(port)
-                # We spawn the processes for rank-1 and above and use the main
-                # process for rank-0 so that we can request debug session
-                # for the main process. We need to do this because the debug
-                # session cannot be started in a subprocess.
-                context = mp.spawn(
-                    _training_worker_helper,
+                processes = mp.spawn(
+                    training_worker,
                     args=(world_size, conf_file, root_dir, paras_queue),
-                    join=False,
-                    nprocs=world_size - 1,
+                    join=True,
+                    nprocs=world_size,
                     start_method='spawn')
-                training_worker(0, world_size, conf_file, root_dir,
-                                paras_queue)
-                context.join()
+        except KeyboardInterrupt:
+            pass
+        except Exception as e:
+            # ``e`` has been printed in the subprocess, so here we won't print it
+            # again. But we raise another error so that we will have a correct
+            # exit code for the program.
+            raise ChildProcessError(f'Training failed on subprocess exception')
+        
+    elif FLAGS.distributed == 'multi-node-multi-gpu':
```

Comment:
Can you add documentation on how multi-node-multi-gpu mode is supposed to be used, i.e. how to start a multi-node-multi-gpu job?

And how do the processes on different machines know how to communicate with the other machines?